tagging_accuracy_report.py

```diff
@@ -0,0 +1,118 @@
+        self.automated_tags_bucket_name = automated_tags_bucket
+        aws_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID", None)
+        aws_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY", None)
+        if aws_access_key_id is None or aws_secret_access_key is None:
+            print("No AWS credentials set in environment")
+            sys.exit(1)
+        self.session = boto3.session.Session(
+            aws_access_key_id, aws_secret_access_key)
+        self.s3_resource = self.session.resource(
+            's3', config=botocore.client.Config(signature_version='s3v4'))
+        self.stemmer = PorterStemmer()
+        self.ecosystem = ecosystem
+
+    def load_manual_tags_json(self, manual_tags_filename):
+        manual_tags_file = self.s3_resource.Object(
+            self.manual_tags_bucket_name, manual_tags_filename
+        ).get()['Body'].read().decode('utf-8')
+        manual_tags_json = json.loads(manual_tags_file)
+        for tag_data in manual_tags_json:
+            if tag_data['ecosystem'] == self.ecosystem:
+                self.package_topic_json = tag_data['package_topic_map']
+                break
+
+    def get_automated_tags_for_package(self, package_name):
+        tags_file_path = "{}/{}.json".format(self.ecosystem, package_name)
+        automated_tags_file = self.s3_resource.Object(
+            self.automated_tags_bucket_name, tags_file_path
+        ).get()['Body'].read().decode('utf-8')
+        automated_tags_json = json.loads(automated_tags_file)
+        return automated_tags_json
+
+    def match_tags(self):
+        no_tags = man_tags_not_in_result = man_tags_not_collected = 0
+        for package_name in self.package_topic_json:
+            # print("Package: " + package_name)
+            manual_tags = set(self.package_topic_json[package_name])
+            # print(manual_tags)
+            try:
+                automated_tags = self.get_automated_tags_for_package(
+                    package_name)
+            except ClientError:
+                print("No tags collected for " + package_name)
+                no_tags += 1
+                continue
+            if 'tags' not in automated_tags \
+                or len(automated_tags['tags']) == 0 \
+                    or len(automated_tags['tags']['_sorted']) == 0:
+                    print(
+                        "No tags available for package {}".format(package_name)
+                    )
+                    no_tags += 1
+                    continue
+            # First check if all manual tags are present in result
+            result = set()
+            for tag_data in automated_tags['tags']['result']:
+                tag_name, tag_score = tag_data
+                result.add(tag_name)
+            # print(result)
+            tags_not_found = manual_tags - result
+            if len(tags_not_found) > 0:
+                print(
+                    """{} not found in result for package {} in automated"""
```

Comment:
You should print these results as `json` or whatever is computer readable. Not sure whether this kind output will be easily parsable.

Reply:
I wasn't sure whether we would want this version of the script in a pipeline so I just let it be more "human friendly" for now, dumping all relevant data to a JSON is definitely an option.

Reply:
> I wasn't sure whether we would want this version of the script in a pipeline so I just let it be more "human friendly" for now, dumping all relevant data to a JSON is definitely an option.

Nice, re-usability is great! I think this is good to go, if we would like to push this to production, we would probably use logging for this.

BTW casting a dict to string does not result in a valid JSON (it's stringified Python's dict):
```python
>>> import json
>>> json.loads(str({"foo": "bar"}))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.6/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)

```

Reply:
You are right @fridex, but @rootAvish is using `set` not `dict` here.